### PR TITLE
US-05_search-a-product

### PR DESCRIPTION
**Pull request by Ludovicscelles**
Search bar activation

### CODE REVIEW

**Review by NicosMQ — APPROVED** [review]

File: client/src/Router.jsx
Comment (Alexandre78R) [inline client/src/Router.jsx:1]: Rename all routes Rename all, road names in French

File: client/src/components/BurgerMenu/BurgerMenu.jsx [hunk 1]
```diff
@@ -22,6 +22,10 @@ export default function BurgerMenu() {
     navigate(`/category/${categoryId}`);
   };
 
+  const handleAllProductsClick = () => {
+    navigate("/category/tous-les-produits");
```
Comment (Alexandre78R) [inline client/src/components/BurgerMenu/BurgerMenu.jsx]: toutes les produits

Comment (Alexandre78R) [reply client/src/components/BurgerMenu/BurgerMenu.jsx]: or re name .../produits

File: client/src/components/CategoriesNavBar/CategoriesNavBar.jsx [hunk 2]
```diff
@@ -27,7 +31,13 @@ function CategoriesNavBar() {
   return (
     <div>
```
Comment (Alexandre78R) [inline client/src/components/CategoriesNavBar/CategoriesNavBar.jsx:32]: replace fragement react

File: client/src/components/NavBar/NavBar.jsx [hunk 3]
```diff
@@ -19,11 +19,23 @@ function NavBar() {
...
   }, []);
 
+  const navigate = useNavigate();
+
+  const homePageDirection = () => {
+    navigate("/");
+  };
+
   return (
     <div>
```
Comment (Alexandre78R) [inline client/src/components/NavBar/NavBar.jsx:29]: replace fragement react

File: client/src/pages/Category/CategoryAllProducts.jsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+function CategoryAllProducts() {
+  const location = useLocation();
+  const everyProduct = useLoaderData();
+
+  const filteredProducts = location.state?.filteredDataItems || everyProduct;
+
+
+  return (
+    <div>
```
Comment (Alexandre78R) [inline client/src/pages/Category/CategoryAllProducts.jsx:14]: replace fragement react

File: client/src/components/AllProducts/AllProducts.jsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import "./AllProducts.css";
+import PropTypes from "prop-types";
+
+function AllProducts({ data }) {
+  return (
+    <div>
```
Comment (Alexandre78R) [inline client/src/components/AllProducts/AllProducts.jsx]: replace fragement react

**Review by Alexandre78R — CHANGES_REQUESTED** [review]

**Pull request closed without merging**